### PR TITLE
Fixed returnNothingOnEmptyPropertyFlags

### PR DESCRIPTION
**Pull request by ksuprynowicz**
Big thanks to @HifiExperiments for spotting this!
Current version will behave incorrectly in some cases, so this is pretty high priority to merge.

### CODE REVIEW

File: libraries/entities/src/EntityItemProperties.cpp [hunk 1]
```diff
@@ -1559,7 +1559,7 @@ ScriptValue EntityItemProperties::copyToScriptValue(ScriptEngine* engine, bool s
 
     const bool pseudoPropertyFlagsActive = pseudoPropertyFlags.test(EntityPseudoPropertyFlag::FlagsActive);
     // Fix to skip the default return all mechanism, when pseudoPropertyFlagsActive
-    const bool returnNothingOnEmptyPropertyFlags = pseudoPropertyFlagsActive && _desiredProperties.isEmpty();
+    const bool returnNothingOnEmptyPropertyFlags = pseudoPropertyFlagsActive;
```
Comment (HifiExperiments) [inline libraries/entities/src/EntityItemProperties.cpp:1562]: does redefining this serve any purpose?  or could we just use pseudoPropertyFlagsActive directly everywhere?

**Review by HifiExperiments — APPROVED** [review]
nice now I get it 😁

File: libraries/entities/src/EntityItemProperties.cpp (hunk 1 above)
Comment (ksuprynowicz) [reply libraries/entities/src/EntityItemProperties.cpp:1562]: It's exactly the same. I decided to redefine it only for readability purposes. Macros would look confusing with pseudoPropertyFlagsActive.

**Comment by ksuprynowicz** [comment]
I tested it and everything works fine :)

**Pull request merged**